### PR TITLE
Implemented functionality to encrypt open text properties in remrem.

### PR DESCRIPTION
**Pull request by zburswa**
Added functionality to decrypt the encrypted properties which are passed from application.properties file.

### CODE REVIEW

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 1]
```diff
@@ -120,8 +123,12 @@ private void readSpringProperties() {
      * @param map  RabbitMq instances map object.
      */
     private void populateRabbitMqConfigurationsBasedOnCatalinaProperties(Map<String, Object> map) {
+        String jasyptSecretPassword = "";
         for (Entry<String, Object> entry : map.entrySet()) {
-            String key = entry.getKey();
+            final String key = entry.getKey();
```
Comment (xvinosi-github) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: find the jasypt away from the loop.

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 2]
```diff
@@ -120,8 +123,11 @@ private void readSpringProperties() {
      * @param map  RabbitMq instances map object.
      */
     private void populateRabbitMqConfigurationsBasedOnCatalinaProperties(Map<String, Object> map) {
+        String jasyptSecretPassword = "";
+        jasyptSecretPassword = (String) map.get("jasypt.encryptor.password");
+        log.info("REMREM JASYPT ENCRYPTION:::::::The secret key passed to decrypt"+jasyptSecretPassword);
```
Comment (a-breid-ericsson) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: Please do not log the "jasyptSecretPassword" No secrets should be logged.

**Comment by zburswa** [comment]
Hi,


The Glue team is currently testing the implemented changes. They were facing some issues while testing. So we have added loggers so that it helps us to know exactly where the issue lies.


I will revert back the loggers once the testing is done.


Thanks & Regards,
Swathi Burada
Tata Consultancy Services
Mail: swathi.burada@tcs.com
Mobile: 9493166145
________________________________

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## 1.0.3
+- Implemented functionality in publish-common to support open text encryption for remrem.
```
Comment (xvinosi-github) [inline CHANGELOG.md]: while reading from properties from catalina.properties.

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 4]
```diff
@@ -120,8 +123,10 @@ private void readSpringProperties() {
      * @param map  RabbitMq instances map object.
      */
     private void populateRabbitMqConfigurationsBasedOnCatalinaProperties(Map<String, Object> map) {
+        String jasyptSecretPassword = "";
+        jasyptSecretPassword = (String) map.get("jasypt.encryptor.password");
```
Comment (xvinosi-github) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: both can be in single line.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+## 1.0.3
+- Implemented functionality in publish-common to support open text encryption for remrem when properties are read from application.properties.
```
Comment (SantoshNC68) [inline CHANGELOG.md]: Fixed broken application properties while encrypting/decrypting passwords.

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/DecryptionUtils.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+import org.jasypt.encryption.pbe.StandardPBEStringEncryptor;
+import org.slf4j.LoggerFactory;
+
+import ch.qos.logback.classic.Logger;
+
+/**
+ *
+ * This class is used for property decryption.
```
Comment (SantoshNC68) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/DecryptionUtils.java]: This utility class is used for decryption of any encrypted property.

**Pull request merged**